### ISSUE TITLE
Add React-compatible update function to useMMKVStorage hook

### ISSUE THREAD

**Issue by SaltedBlowfish** [ammarahm-ed/react-native-mmkv-storage#98]
_See https://github.com/ammarahm-ed/react-native-mmkv-storage/issues/13#issuecomment-841727927 for the original mention that led to this issue being opened_

Currently, the `useMMKVStorage` hook doesn't reflect the expectation of the "setter" portion of the React built-in state hooks. It's not documented that it should, but would provide convenience for developers implementing this package...

Ideally, we would be able to have a state change with a previous value provided, similarly to the `React.useState` hook for making sure value writes are happening without overwriting previous states:

### React's API

```
const [value, setValue] = React.useState(0);
...
setValue((prev) => {
  // we have access to prev in here for making safe changes to the state
  return prev + 1
})
```

### RN MMKV's API

```
const MMKV = new MMKVStorage.Loader().initialize();
const [value, setValue] = useMMKVStorage('counter', MMKV);
...
setValue(
  // cannot be a function that accepts previous state
)
```

To support this, the `setValue` function above must support a callback as a parameter for dynamically computing the next state.

Usage example:

```
const MMKV = new MMKVStorage.Loader().initialize();
const [value, setValue] = useMMKVStorage<number>('counter', MMKV);
...
setValue((prev) => {
  return prev === undefined ? 0 : prev + 1
})
```

**Comment by ammarahm-ed** [comment]
Hey @SaltedBlowfish do you plan to work on sending a PR with this feature?

**Comment by ammarahm-ed** [comment]
@SaltedBlowfish I have added support for setter. test it out on master so I can release the next version

**Comment by SaltedBlowfish** [comment]
@ammarahm-ed Sorry for the delay. I did plan on working on this but I'm traveling right now and ended up not having the time. I didn't realize you'd get to it so quickly. I won't be able to test it out for another week or so though.

**Comment by SaltedBlowfish** [comment]
@ammarahm-ed I ended up pulling your changes and resolving conflicts with my fork. See my PR here: https://github.com/ammarahm-ed/react-native-mmkv-storage/pull/103

I have not tested it yet (like I said I'm traveling right now).

**Comment by ammarahm-ed** [comment]
Shipped in v0.5.9
